docs/source/_templates/txn_submit_tutorial.rst

```diff
@@ -0,0 +1,484 @@
+
+{% else %}
+{# Python code is the default #}
+
+The Python *secp256k1* module provides a *PrivateKey* handler class from which
+we can generate the actual bytes to use for a key.
+
+.. code-block:: python
+
+    import secp256k1
+
+    key_handler = secp256k1.PrivateKey()
+    private_key_bytes = key_handler.private_key
+
+{% endif %}
+
+.. note::
+
+   This key is the **only** way to prove your identity on the blockchain. Any
+   person possessing it will be able to sign Transactions using your identity,
+   and there is no way to recover it if lost. It is very important that any
+   private key is kept secret and secure.
+
+In addition to a private key, you will need a shareable public key generated
+from the private key. It should be encoded as a hexadecimal string, to
+distribute with the Transaction and confirm that its signature is valid.
+
+{% if language == 'JavaScript' %}
+
+.. code-block:: javascript
+
+    const publicKeyBytes = secp256k1.publicKeyCreate(privateKeyBytes)
+
+    const publicKeyHex = publicKeyBytes.toString('hex')
+
+{% else %}
+
+.. code-block:: python
+
+    public_key_bytes = key_handler.pubkey.serialize()
+
+    public_key_hex = public_key_bytes.hex()
+
+{% endif %}
+
+
+{% include 'partials/encoding_your_payload.rst' %}
+
+
+Building the Transaction
+========================
+
+*Transactions* are the basis for individual changes of state to the Sawtooth
+blockchain. They are conmposed of a binary payload, a binary-encoded
+*TransactionHeader* with some cryptographic safeguards and metadata about how it
+should be handled, and a signature of that header. It would be worthwhile to
+familiarize yourself with the information in
+:doc:`/architecture/transactions_and_batches`, pararticularly the definition of
+TransactionHeaders.
+
+
+1. Create a SHA-512 Payload Hash
+--------------------------------
+
+However the payload was originally encoded, in order to confirm it has not been
+tampered with, a hash of it must be included within the Transaction's header.
+This hash should be created using the SHA-512 function, and then formatted as a
+hexadecimal string.
+
+{% if language == 'JavaScript' %}
+
+.. code-block:: javascript
+
+    const sha512 = require('crypto').createHash('sha512')
+
+    const payloadSha512 = sha512.update(payloadBytes).digest('hex')
+
+{% else %}
+
+.. code-block:: python
+
+    from hashlib import sha512
+
+    payload_sha512 = sha512(payload_bytes).hexdigest()
+
+{% endif %}
+
+
+2. Build the TransactionHeader
+------------------------------
+
+Transactions and their headers are built using
+`Google Protocol Buffer <https://developers.google.com/protocol-buffers/>`_
+(or Protobuf) format. This allows data to be serialized and deserialzed
+consistently and efficiently across multiple platforms and multiple languages.
+The Protobuf definition files are located in the
+`/protos <https://github.com/hyperledger/sawtooth-core/tree/master/protos>`_,
```

Comment:
extra comma? The Protobuf definition files are located in the
 +`/protos <https://github.com/hyperledger/sawtooth-core/tree/master/protos>`_, 

The comma after the underscore at end there doesn't seem wanted in the rendered text.